hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py

```diff
@@ -1,33 +1,96 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
 import json
+import typing as t
 
 from hmalib.common.logging import get_logger
 from hmalib.models import MatchMessage, Label
 
 logger = get_logger(__name__)
 
-
 def lambda_handler(event, context):
     """
-    This lambda is called once per match per dataset. If a single hash matches
-    multiple datasets, this will be called multiple times.
-
-    Eventually, this will be just an action labeller. It will label a match
-    record with the action it recommends. A separate system will be stood up
-    that aggregates labels and 'decides' which action would be taken.
-
-    For now, this method will,
-    - it will construct the MatchMessage object and identify the actions it
-      needs to invoke.
-    - for now, rather than fanning out to individual specific lambdas, it will
-      call all specific functions serially!
+    This lambda is called when one or more matches are found. If a single hash matches
+    multiple datasets, this will be called only once.
+
+    Action labels are generated for each match message, then an action is performed
+    corresponding to each action label.
     """
-    for sqs_record in event["Records"]:
+    for sqs_record in event["Records"]: #confused about this... are we popping off more than one thing at a time from the "matches" queue?
         sns_notification = json.loads(sqs_record["body"])
         match_message: MatchMessage = MatchMessage.from_sns_message(
             sns_notification["Message"]
         )
+        action_labels = get_action_labels(match_message)
+        for action_label in action_labels:
+            perform_action(match_message, action_label)
+
+        # the more I write code here about reacting to ThreatExchange, the more I wish it was somewhere else entirely. Implementing
+        # here forever ties up action-related code with ThreatExchange-specific code / functionality. I prefer putting
+        # ThreatExchange-related specifics for reacting into another lambda that also subscribes to the matcher's notifications.
+        if threat_exchange_reacting_is_enabled():
+            threat_exchange_reaction_labels = get_threat_exchange_reaction_labels(action_labels)
+            if threat_exchange_reaction_labels is t.List[Label]:
```

Comment:
This probably would not work. `python` still does not have types. You can only check whether something is a list, not whether the list has a specific type.

Reply:
This is trying to check if all items in the list are Labels? if so the pythonic way would be...
```
if all( label is Label for label in threat_exchange_reaction_labels ):
    ...
```

Reply:
@schatten My need / intent is to validate (in a python-y way) that there's something to iterate over, in case there's a concept of null or undefined in Python... I don't want to do the `for` if there isn't a list... But this is a general thing that I don't know anything about yet in Python. Maybe I can just remove line 33? Or should I leave it the way it is but remove `Label`:

`if threat_exchange_reaction_labels is t.List[]:`

Reply:
Oh you want to check if the list is empty?

either:
```
if len(threat_exchange_reaction_labels) > 0 :
```

or just :
```
if threat_exchange_reaction_labels :
```